### ISSUE TITLE
bug: fix latency in offchain publishing

### ISSUE THREAD

**Issue by EvolveArt** [astraly-labs/pragma-node#83]
**Current Behavior**

One cannot publish more than every ~2s due to some latency in our publish endpoint.

**Expected Behavior**

We should find where this latency comes from and fix it to allow publishing every 500ms.

**Comment by akhercha** [comment]
## Investigations results

The `pragma-node` has been checked and we measured the publish steps. We identified that the main bottleneck was in fact coming from our `pragma-sdk` python SDK, when the publisher signs the whole batch of entries he wants to publish.

More specifically, the `message_hash` and `sign_message` used [here](https://github.com/astraly-labs/pragma-sdk/blob/master/pragma-sdk/pragma_sdk/offchain/signer.py):

![image](https://github.com/user-attachments/assets/20969226-d649-42cc-9e89-8a455bb72f6b)

AFAIK, this is normal behavior as the hash function scales with the number of elements that needs to be hashed.


_____

However, the main issue encountered from our publishers was when publishing ~100 entries, and for this use case the delay is ~0.4s:
![image](https://github.com/user-attachments/assets/4d46e6e0-000f-402f-9f20-b82ed3a0cc32)

We want it to be below 0.5s so this is good.

We will publish the new SDK version and let our publishers test again the publish process - if their delay is for some reasons greater than 0.5s we will dig more on how to optimize the hash process - maybe by using another hash? To be determined.

**Comment by akhercha** [comment]
**TLDR**:  `starknet.py` version 0.23 allows us to use a poseidon hash instead of pedersen. This would increase our performances by ~60%. To be done if needed. This would mean we have to update our check signature method in `pragma-node`.

_______

In their latest release, `starknet.py` allows to update the field `revision` of a TypedData:
```python
    @property
    def _hash_method(self) -> HashMethod:
        if self.domain.resolved_revision == Revision.V0:
            return HashMethod.PEDERSEN
        return HashMethod.POSEIDON
```

We were using `V0`, i.e. the pedersen hash in version 0.22.
What we can do by upgrading `starknet.py` to 0.23 is update this:

```python
message = {
        "domain": {"name": "Pragma", "version": "1"},
        "primaryType": "Request",
        "message": {
            "action": "Publish",
            "entries": Entry.serialize_entries(entries),
        },
        "types": {
            "StarkNetDomain": [
                {"name": "name", "type": "felt"},
                {"name": "version", "type": "felt"},
            ],
# ...
```

by:

```python
message = {
        "domain": {"name": "Pragma", "version": "1", "chainId": "1", "revision": "1"},
        "primaryType": "Request",
        "message": {
            "action": "Publish",
            "entries": Entry.serialize_entries(entries),
        },
        "types": {
            "StarknetDomain": [
                {"name": "name", "type": "felt"},
                {"name": "version", "type": "felt"},
                { "name": "chainId", "type": "felt" },
                { "name": "revision", "type": "felt" }
            ],
# ...
```
We added `chainId` & `revision`, two new mandatory fields. If we set the `revision` field to 1 as shown above, we will switch to using the `POSEIDON` hash method, which is 60% faster.

### Example when publishing 1000 entries

- (**what we have now**) using the `PEDERSEN` hash:
![image](https://github.com/user-attachments/assets/0782ec5f-26e7-4607-95ee-24d35f3c0ec6)

- (**what we would have**) using the `POSEIDON` hash:
![image](https://github.com/user-attachments/assets/11dff93a-b28d-408d-8ec8-0a8880cbafbe)
